src/vsg/state/DescriptorBuffer.cpp

```diff
@@ -177,15 +178,16 @@ void DescriptorBuffer::compile(Context& context)
                 if (bufferInfo->buffer->getDeviceMemory(deviceID) == nullptr)
                 {
                     auto memRequirements = bufferInfo->buffer->getMemoryRequirements(deviceID);
-                    auto memory = vsg::DeviceMemory::create(context.device, memRequirements, VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT);
-                    auto [allocated, offset] = memory->reserve(bufferInfo->buffer->size);
-                    if (allocated)
+                    auto [deviceMemory, offset]
+                        = context.deviceMemoryBufferPools->reserveMemory(memRequirements,
+                                                                         VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT);
+                    if (deviceMemory)
                     {
-                        bufferInfo->buffer->bind(memory, offset);
+                        bufferInfo->buffer->bind(deviceMemory, offset);
                     }
                     else
                     {
-                        warn("DescriptorBuffer::compile(..) unable to allocate buffer within associated DeviceMemory.");
+                        throw Exception{"Error: DescriptorBuffer::compile(..) unable to allocate buffer."};
```

Comment:
If we are to throw an exception then the returning a relevant Vulkan error would be helpful.

Alternatively we could use vsg::fatal().

Reply:
> If we are to throw an exception then the returning a relevant Vulkan error would be helpful.
> 
I agree. That would require changing the memory pool interface,  but perhaps that is desirable.
> Alternatively we could use vsg::fatal().

Reply:
This close to a release I want to avoid getting diverted into lots of other work - occasionally I need to get some paid work done :-)